alkos/kernel/src/init.cpp

```diff
@@ -5,36 +5,17 @@
 #include <terminal.hpp>
 #include <debug.hpp>
 
-/* external init procedures */
-extern "C" void enable_osxsave();
-
-extern "C" void enable_sse();
-
-extern "C" void enable_avx();
-
 extern "C" void PreKernelInit() {
-    /**
-     * TODO:
-     * 1. Enable interrupts
-     * 2. GDT, IDT, TSS, etc.
-     */
-
     TerminalInit();
     TRACE_INFO("Starting pre-kernel initialization...");
-    __stack_chk_init();
 
-    /* NOTE: sequence is important */
     TRACE_INFO("Starting to setup CPU features...");
-    TRACE_INFO("Setting up OS XSAVE...");
```

Comment:
Again, i believe this should be in boot64, called before kernel.
The kernel should be run with maximum amount of arch dependent things already done. It also stinks to me that arch dependend setups are called inside the kernel.

The terminal is a bit of a different case - since there is no other way of implementation than providing an abi for drivers,  and it needs be used in later places of the code. but for thing like init where it's done once and forgot - its a different case.

Reply:
Will be addressed on some sync call.